lib/donation_system/salesforce/donation_validator.rb

```diff
@@ -46,12 +46,18 @@ def validate_data
       end
 
       def validate_amount
-        :invalid_amount unless data&.amount && !data.amount.to_i.zero?
+        :invalid_amount unless data&.payment_data&.amount && integer?
```

Comment:
It would be clearer to to pass `data.payment_data.amount` to `integer?` as an argument, rather than having a method called `integer?` that is coupled to `data.payment_data.amount`

Or instead, rename `integer?` to `amount_is_integer?` ?

Reply:
I'll rename it. It doesn't make sense to pass an instance attribute as an argument of an instance method.